learning_assistant/views.py

```diff
@@ -91,3 +91,41 @@ def post(self, request, course_id):
         status_code, message = get_chat_response(prompt_template, message_list, course_id)
 
         return Response(status=status_code, data=message)
+
+
+class LearningAssistantEnabledView(APIView):
+    """
+    View to retrieve whether the Learning Assistant is enabled for a course.
+
+    This endpoint returns a boolean representing whether the Learning Assistant feature is enabled in a course
+    represented by the course_key, which is provided in the URL.
+
+    Accepts: [GET]
+
+    Path: /learning_assistant/v1/course_id/{course_key}/enabled
+
+    Parameters:
+        * course_key: the ID of the course
+
+    Responses:
+        * 200: OK
+    """
+
+    authentication_classes = (SessionAuthentication, JwtAuthentication,)
+    permission_classes = (IsAuthenticated,)
+
+    def get(self, request, course_id):
+        """
+        Given a course ID, retrieve whether the Learning Assistant is enabled for the corresponding course.
+
+        The response will be in the following format.
+
+            {'enabled': <bool>}
+        """
+        courserun_key = CourseKey.from_string(course_id)
```

Comment:
[nit] is there a reason to use `courserun_key` rather than course_key here? I like it if it is more specific but I wasn't sure if these are just the same? If so, I might suggest renaming to simplify.

Reply:
I copied some of this code from the view about, which Alie wrote, but I think there is a reason. An instance of a `CourseKey` technically represents a run of a course, or a course run. There is a difference between a canonical or "shell" course and the actual instance of the course, which is a course run. I think, in most places, we just use `course_key` to refer to `CourseKey` or the course run key, because it's simpler and an implementation detail. The reason we're being a little more explicit here is that we actually do want the course key in some places.

The Xpert index that we use to pull data from on the Xpert Platform backend expects a course key. That is, they do not want the course run key. So we're handling course run keys and course keys in this code base, which is why I think Alie chose a more specific name. You can see where we "convert" a course run key into a course key [here](https://github.com/edx/learning-assistant/blob/8a25061b1afc855f1f578965797bb7c4864eb08e/learning_assistant/utils.py#L63).

If the use of `courserun_key` is confusing here, I could change it to `course_key`. I don't think the distinction in this view is super important.

Reply:
That makes sense! Thanks for clarifying! I'm good with keeping it as is, as like you said, it's a little more explicit. Resolving now.